keras_cv/models/object_detection/retinanet/retinanet.py

```diff
@@ -193,7 +192,7 @@ def __init__(
         batch_size = tf.shape(images)[0]
         cls_pred = []
         box_pred = []
-        for feature in features:
+        for feature in features.values():
```

Comment:
@ianstenbit Let me know if you want me to handle the output of the FeaturePyramid differently. I wasn't sure what would be the best way to handle changing the output to a dict and decided to go with this option because of how it is handled in the [yolo_v8_detector](https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py#L409). I believe dicts are ordered in Python3.6 and beyond, so this should be deterministic.

Reply:
Yes, as of Python3.7 I think, dicts are guaranteed to be insertion-ordered.

That said, it may be nice to be a bit more overt where we can, e.g. by saying something like:
```
for pyramid_level in ["P1", "P2", ...]:
    feature = features[pyramid_level]
    ...
```

Reply:
Done